text/0000-build-std/5-stage-1b.md

```diff
@@ -0,0 +1,986 @@
+[dev-dependencies-and-build-dependencies]: #dev-dependencies-and-build-dependencies
+
+Explicit dependencies on the standard library are not supported in
+`build-dependencies` ([?][rationale-no-deps-in-build-deps]).
+
+Implicit and explicit dependencies on the standard library are supported for
+`dev-dependencies` in the same way as regular `dependencies`.
+
+## Registries
+[registries]: #registries
+
+Standard library dependencies will be present in the registry index
+([?][rationale-cargo-index]). A `builtin_deps` key is added to the
+[index's JSON schema][cargo-json-schema] ([?][rationale-cargo-builtindeps]).
+`builtin_deps` is similar to the existing `deps` key and contains a list of JSON
+objects, each representing a dependency that is "builtin" to the Rust toolchain
+and cannot otherwise be found in the registry.
+
+> [!NOTE]
+>
+> It is expected that the keys of these objects will be:
+>
+> - `name`
+>   - String containing name of the `builtin` package. Can shadow the names of
+>     other packages in the registry (except those packages in the `deps` key
+>     of the current package) ([?][rationale-cargo-index-shadowing])
+>
+> - `features`:
+>   - An array of strings containing enabled features in order to support changing
+>     the standard library features on nightly. Optional, empty by default.
+>
+> - `optional`, `default_features`, `target`, `kind`:
+>   - These keys have the same definition as in the `deps` key.
+>
+> The keys `req`, `registry` and `package` from `deps` are not required per the
+> limitations on builtin dependencies.
+>
+> The key is optional and its default value will be the implicit builtin
+> dependencies:
+>
+> ```json
+> "builtin_deps" : [
+>     {
+>         "name": "std",
+>         "features": [],
+>         "optional": false,
+>         "default_features": true,
+>         "target": null,
+>         "kind": "normal",
+>     },
+>     {
+>         "name": "alloc",
+>         ... # as above
+>     },
+>     {
+>         "name": "core",
+>         ... # as above
+>     }
+> ]
+> ```
+
+*See the following sections for rationale/alternatives:*
+
+- [*Why add standard library crates to Cargo's index?*][rationale-cargo-index]
+- [*Why add a new key to Cargo's registry index JSON schema?*][rationale-cargo-builtindeps]
+- [*Why can `builtin_deps` shadow other packages in the registry?*][rationale-cargo-index-shadowing]
+
+## Cargo subcommands
+[cargo-subcommands]: #cargo-subcommands
+
+As opaque dependencies, any Cargo command which accepts a package spec with `-p`
+will only additionally recognise `core`, `alloc` and `std` and none of their
+dependencies. Many of Cargo's subcommands will need modification to support
+build-std:
+
+[`cargo add`][cargo-add] will add `core`, `alloc` or `std` explicitly to the
+manifest if invoked with those crate names:
+
+```toml
+[package]
+name = "hello_world"
+version = "0.1.0"
+edition = "2024"
+
+[dependencies]
+std = { builtin = true } # <-- this would be added
+```
+
```

Comment:
So `cargo add std` will change from adding a path dependency on `std` to adding it builtin?

Or do we want a `--builtin` flag.  The flag seems redundant but it makes things less special and it allows depending on additional builtins on nightly

Reply:
Yeah I think this is the right thing to do: https://github.com/davidtwco/rfcs/pull/2/commits/af14747b86a96494d6ab9c5ca04118daf40a768d
 A user will still get an error message when doing `cargo add std` since it cant be found on crates.io, and maybe we could add a help to that error

Reply:
Elaborated on this more in 725065eac08bd38c83bc2318880fa473f29b3c8c